extensions/html-language-features/server/src/modes/javascriptMode.ts

```diff
@@ -601,3 +611,11 @@ function generateIndent(level: number, options: FormattingOptions) {
 		return repeat('\t', level);
 	}
 }
+
+function getLibDomUriAndDocument(): [string, TextDocument] {
```

Comment:
To compute the library location, add a new function to https://github.com/microsoft/vscode/blob/91e6ae971de19ccfb32e90f064cf194d81a29a39/extensions/html-language-features/server/src/modes/javascriptLibs.ts#L11 and reuse the logic there.

As for the document URI, this is trickier as this also needs to run on the web (vscode.dev), and there's no 'file' there.
You probably need to start from the [extensionUri](https://github.com/microsoft/vscode/blob/91e6ae971de19ccfb32e90f064cf194d81a29a39/src/vscode-dts/vscode.d.ts#L8269) that you get on the HTML client, and pass it on the server.